src/System.Data.Common/tests/System/Data/SqlTypes/SqlDecimalTest.cs

```diff
@@ -276,6 +276,52 @@ public void AdjustScale()
         }
 
         [Fact]
+        public void AdjustScaleForFlatZero()
```

Comment:
Can you change the name of this test to "AdjustScaleForZero"? We're testing more than just the "0" case here.